File: packages/agent/src/ast/languages/fixtures/sample.cs
using System;
using System.Collections.Generic;

namespace Demo;

/// <summary>Greeter</summary>
public class Greeter : IGreeter
{
	private int count;

	public string Name { get; set; } = "";

	public Greeter() {}

	public Greeter(string name)
	{
		Name = name;
	}

	public void Hello()
	{
		count += 1;
	}

	public event EventHandler? Done;

	public string this[int index] => Name;
}

internal struct Box
{
	public int Value;
}

public interface IGreeter
{
	void Hello();
}

public enum Color
{
	Red,
	Green,
}

public record Person(string Name);

public delegate void Notify(string message);

static class Helpers
{
	public static IEnumerable<int> Range(int n) => System.Linq.Enumerable.Range(0, n);
}

File: packages/agent/src/ast/languages/fixtures/sample.swift
import Foundation

/// Greeter type
public class Greeter: Named {
	public var name: String = ""

	public init() {}

	public init(name: String) {
		self.name = name
	}

	public func hello() {
		print("hi \(name)")
	}

	deinit {}
}

extension Greeter {
	public func shout() {
		print(name.uppercased())
	}
}

public protocol Named {
	var name: String { get }
	func nameLabel()
}

internal struct Box {
	var value: Int
}

private enum Color {
	case red
	case green
}

public typealias Alias = Greeter

public enum Result {
	case ok(Int)
	case err(String)
}
